# Lab 3: KNN, Bayes Classifier and Decision Trees

model just needs the data not the footer and header
nota stratify

In [ ]:
import numpy as np
import pandas as pd
import seaborn as sns

from sklearn.datasets import fetch_20newsgroups
newsgroups = fetch_20newsgroups(subset='all')


In [20]:
print(type(newsgroups))
print(newsgroups.keys())

<class 'sklearn.utils._bunch.Bunch'>
dict_keys(['data', 'filenames', 'target_names', 'target', 'DESCR'])


### 1. Loading and exploring data
- a) How many instances (examples) and variables are in the dataset?

In [ ]:
print(len(newsgroups.data))

18846


There are 18846 instances in this dataset, but it has no variables. The dataset is just composed of raw strings just a single entry of text

In [28]:
print(newsgroups.target[1])
print(newsgroups.target_names[newsgroups.target[1]])

3
comp.sys.ibm.pc.hardware
